### PR TITLE
Ensure that primitive types in @ConfigProperties work as expected

### LINKED ISSUE #13523: @ConfigProperties doesn't handle unset primitives
**Describe the bug**
When primitives are used in a class annotated with `@ConfigProperties`, then Quarkus fails at runtime if the primitive is not set.

**Expected behavior**
Unset primitives should not prevent Quarkus from starting and their value should just be the default set by the JVM

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #13523

### CODE REVIEW

File: extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/ClassConfigPropertiesUtil.java [hunk 1]
```diff
@@ -420,8 +420,10 @@ private static void populateTypicalProperty(MethodCreator methodCreator, ResultH
             createWriteValue(methodCreator, configObject, field, setter, useFieldAccess, setterValue);
 
         }
-        configPropertyBuildItemCandidates
-                .add(new ConfigPropertyBuildItemCandidate(field.name(), fullConfigName, fieldType));
+        if (field.type().kind() != Type.Kind.PRIMITIVE) { // the JVM assigns primitive types a default even though it doesn't show up in the bytecode
+            configPropertyBuildItemCandidates
+                    .add(new ConfigPropertyBuildItemCandidate(field.name(), fullConfigName, fieldType));
```
Comment (gsmet) [inline extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/ClassConfigPropertiesUtil.java:425]: Hmmm, and we are sure that nothing that `ConfigPropertyBuildItemCandidate` triggers will ever be needed by primitives?

Comment (geoand) [reply extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/ClassConfigPropertiesUtil.java:425]: Yup, that list is only used in turn to populate `ConfigPropertyBuildItem` (which are the user properties that are required when booting the application).
The `getPropertyType` field of `ConfigPropertyBuildItem` is only in order to register for reflection which if of course is not needed for primitives.
So there is absolutely no reason that these primites should ever end up in the candidates list.

**Review by gsmet — APPROVED** [review]

File: extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/ClassConfigPropertiesUtil.java (hunk 1 above)
Comment (gsmet) [reply extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/ClassConfigPropertiesUtil.java:425]: OK, thanks for the quick turnaround.

Comment (geoand) [reply extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/ClassConfigPropertiesUtil.java:425]: You are welcome!

**Pull request merged**